packages/trader/src/Modules/Contract/Components/LastDigitPrediction/last-digit-prediction.jsx

```diff
@@ -57,82 +71,65 @@ class LastDigitPrediction extends React.Component {
         return barrier_map[contract_type](num) ? num : null;
     };
 
-    get offset() {
-        return isMobile() ? this.digit_offset_mobile : this.digit_offset;
-    }
+    const getOffset = isMobile() ? digit_offset_mobile : digit_offset;
 
-    get is_selectable_digit_type() {
-        return isMobile() ? this.props.trade_type !== 'even_odd' : false;
-    }
+    const isSelectableDigitType = () => (isMobile() ? trade_type !== 'even_odd' : false);
 
-    render() {
-        const {
-            digits,
-            digits_info,
-            is_digit_contract,
-            has_entry_spot,
-            is_ended,
-            is_trade_page,
-            tick,
-            selected_digit,
-            status,
-        } = this.props;
-        const digits_array = Object.keys(digits_info)
-            .sort()
-            .map(spot_time => digits_info[spot_time]);
-        const last_contract_digit = digits_array.slice(-1)[0] || {};
+    const digits_array = Object.keys(digits_info)
+        .sort()
+        .map(spot_time => digits_info[spot_time]);
+    const last_contract_digit = digits_array.slice(-1)[0] || {};
 
-        // 'won' or 'lost' status exists after contract expiry
-        const is_won = is_ended && status === 'won' && is_digit_contract;
-        // need to explicitly have is_lost condition here
-        // because negating is_won would always be true,
-        // but we only need is_lost condition only once we have the 'won' or 'lost' status
-        const is_lost = is_ended && status === 'lost' && is_digit_contract;
+    // 'won' or 'lost' status exists after contract expiry
+    const is_won = is_ended && status === 'won' && is_digit_contract;
+    // need to explicitly have is_lost condition here
+    // because negating is_won would always be true,
+    // but we only need is_lost condition only once we have the 'won' or 'lost' status
+    const is_lost = is_ended && status === 'lost' && is_digit_contract;
```

Comment:
Are you comfortable with `is_ended`? :sweat_smile: 
How about `is_finished` or `has_ended`?

Reply:
`is_ended` is from contract store. It's better not to touch it 😄